sdk/tdferrors.go

```diff
@@ -0,0 +1,37 @@
+package sdk
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	errFileTooLarge     = errors.New("tdf: can't create tdf larger than 64gb")
+	errWriteFailed      = errors.New("tdf: io.writer fail to write all bytes")
+	errInvalidKasInfo   = errors.New("tdf: kas information is missing")
+	errKasPubKeyMissing = errors.New("tdf: kas public key is missing")
+	errRewrapForbidden  = errors.New("tdf: rewrap request 403")
+	// Exposed tamper detection errors, Catch all possible tamper errors with errors.Is(ErrTampered)
+	ErrTampered                = errors.New("tamper detected")
+	ErrRootSigValidation       = fmt.Errorf("[%w] tdf: failed integrity check on root signature", ErrTampered)
+	ErrSegSizeMismatch         = fmt.Errorf("[%w] tdf: mismatch encrypted segment size in manifest", ErrTampered)
+	ErrTDFReaderFailed         = fmt.Errorf("[%w] tdf: fail to read bytes from TDFReader", ErrTampered)
+	ErrSegSigValidation        = fmt.Errorf("[%w] tdf: failed integrity check on segment hash", ErrTampered)
+	ErrTDFPayloadReadFail      = fmt.Errorf("[%w] tdf: fail to read payload from tdf", ErrTampered)
```

Comment:
Same here not sure if this indicates tampering.

Reply:
i forget the reasoning for this one, @dmihalcik-virtru ?

Reply:
These indicate that the segment pointers are invalid, e.g. out of sequence, overlapping, or out of range. Likely indicative of a corrupt file in some fashion, although I suppose there could be some technically 'readable' files that have this property, they will not be a valid TDF file.